src/provider/model/FinEnvironment.ts

```diff
@@ -133,14 +133,16 @@ export class FinEnvironment extends AsyncInit implements Environment {
             this.windowClosed.emit(identity);
         });
 
-        // Register windows that were running before launching the FDC3 service
-        const windowInfo = await fin.System.getAllWindows();
+        Injector.initialized.then(async () => {
```

Comment:
We can't await the injector because we are calling this from within an `init` method that the injector calls, so it's a bit circular.  The reason it is needed is because we were running into the issue of window creation logic running before in the model before all of the injectables were ready

Reply:
I've put a comment above this line explaining the await issue

Reply:
Huh. Makes sense. Has something changed that's made this a problem? Or did we just not see this problem before?

Reply:
I think it was masked by the way we were setting the application directory before using the service config loader

Reply:
Couple of paranoid suggestions - 
- Move `const windowInfo = await fin.System.getAllWindows();` outside the `then` block, so that we do this as soon as possible after adding the event listeners (or put it right before we add the event listeners), so we avoid a badly timed window creation being processed twice

- Put `await Injector.initialized` as the first line of the two event listeners, so we guarantee that not signals get dispatched until the service is properly ready

Reply:
Added both